src/bloggy_frontend/src/pages/NewPost.jsx

```diff
@@ -10,45 +10,48 @@ import {
   postValidationSchema,
 } from '../constants/formConstants'
 import { Principal } from '@dfinity/principal'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
 
 const NewPost = () => {
   const { showToast } = useToast()
   const navigate = useNavigate()
   const { artemisAdapter } = useAuth()
+  const queryClient = useQueryClient()
 
-  const addPost = async (values, { setSubmitting, setErrors, resetForm }) => {
-    try {
-      const { title, description } = values
-      if (!artemisAdapter) {
-        throw new Error('Artemis wallet is not connected')
-      }
-      const principalId = artemisAdapter.principalId
-      const principal = Principal.fromText(principalId)
-      console.log('principal:', principal)
-      const result = await bloggy_backend.createPost(
-        title,
-        description,
-        principal
-      )
-      console.log('Result from createPost:', result)
+  const addPost = async (values) => {
+    const { title, description } = values
+    if (!artemisAdapter) {
+      throw new Error('Artemis wallet is not connected')
+    }
+    const principalId = artemisAdapter.principalId
+    const principal = Principal.fromText(principalId)
+    const result = await bloggy_backend.createPost(
+      title,
+      description,
+      principal
+    )
+    return result
+  }
+
+  const mutation = useMutation({
+    mutationFn: addPost,
+    onSuccess: (result) => {
       if (result.hasOwnProperty('ok')) {
         showToast('Post added successfully!', 'success')
-        resetForm()
-        setTimeout(() => {
-          navigate('/posts')
-        }, 2000)
+        queryClient.invalidateQueries('posts')
+        console.log('Post added ', result)
+        navigate('/posts', { replace: true })
```

Comment:
Create a custom hook and move to `src/hook/posts.js` and export this fn as `useCreatePost`